josepholim iP

josepholim opened the pull request:
### Je-O Chatbot

> "Luck is a dividend of sweat. The more you sweat, the luckier you get." - Ray Kroc ([source](https://www.brainyquote.com/quotes/ray_kroc_129827?src=t_work))

Je-O Chatbot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/josepholim/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

josepholim committed:
Level-0

josepholim committed:
Level-1

josepholim committed:
Level-2

josepholim committed:
Level-3

josepholim committed:
Level-4

josepholim committed:
Automated Text UI Testing

josepholim committed:
Level-5

josepholim committed:
Level-6

josepholim committed:
Revise exception

josepholim committed:
Level-7

josepholim committed:
Merge branch 'branch-Level-7'

josepholim committed:
Level-8

josepholim committed:
Merge branch 'branch-Level-8'

josepholim committed:
Revise exception 2

josepholim committed:
Add More OOP

josepholim committed:
Merge branch 'branch-A-MoreOOP'

josepholim committed:
Organize into Packages

josepholim committed:
Organize into Packages

josepholim committed:
Merge branch 'branch-A-Packages'

josepholim committed:
Move Duke.java

josepholim committed:
Merge remote-tracking branch 'origin/add-gradle-support'

josepholim committed:
Use Gradle

josepholim committed:
Add JUnit Tests

josepholim committed:
Merge branch 'branch-A-JUnit'

josepholim committed:
Level-9 Find

josepholim committed:
Add JavaDoc comments

josepholim committed:
Follow the Coding Standard

josepholim committed:
Merge branch 'branch-A-JavaDoc'

josepholim committed:
Resolve merge conflict

josepholim committed:
Resolve merge conflict

josepholim committed:
Add javadocs for find

josepholim committed:
Use Checkstyle

josepholim committed:
Merge branch 'branch-A-Checkstyle'

JizhuoChen commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+ * @author Joseph Oliver Lim
+ */
+public class DeleteCommand extends Command {
+    private int index;
```
Maybe could be more specific in naming parameter 🤔

JizhuoChen commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,53 @@
+ */
+public class EventCommand extends Command {
+    private String description;
+    private String from;
```
Maybe could be more specific in naming parameter 🤔 e.g. `startingTime`

JizhuoChen commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,53 @@
+public class EventCommand extends Command {
+    private String description;
+    private String from;
+    private String to;
```
Maybe could be more specific in naming parameter 🤔e.g. `endingTime`

JizhuoChen commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,111 @@
+ */
+public class Storage {
+    private String directoryString = "./data";
+    private String filePathString = directoryString + "/duke.txt";
```
filePath vs filePathString, which one do you think is better?🤔

JizhuoChen reviewed (commented):
LGTM 💯 :D

josepholim committed:
Add GUI

josepholim committed:
Merge branch 'branch-Level-10'

josepholim committed:
Add new release

josepholim committed:
Add Assertions

josepholim committed:
Improve Code Quality

Further abstract methods which have more than 30 lines of code for
better readability. Then, split the codes into several methods where
applicable and necessary.

Use better names for variables and methods, names that self-define the
purpose of the variables and methods, for better readability.

Rephrase some of the JavaDoc comments so that they are more
understandable.

josepholim committed:
Merge pull request #1 from josepholim/branch-A-Assertions

Add Assertions

josepholim committed:
Merge pull request #2 from josepholim/branch-A-CodeQuality

Improve Code Quality

josepholim committed:
Add CI

Currently testing the application is done manually every time something
is changed.

The testing can be automated in Github using Continuous Integration, so
that is being set up in this commit.

josepholim committed:
Fix DeadlineTest

josepholim committed:
Add a Sort Command to sort the tasks.

The Sort Command is done by entering "sort".
It sorts the task by the type of the task initially,
based on this order: ToDo, Deadline, Event.
Then, for Deadline, it will sort by the deadline date chronologically.
Then, for Event, it will sort by the start date chronologically.
After sorting, the GUI will show the sorted list of tasks.

josepholim committed:
Merge pull request #4 from josepholim/branch-C-Sort

Add a Sort Command to sort the tasks.

josepholim committed:
Fix bug cannot wrap text

josepholim committed:
Merge pull request #5 from josepholim/branch-A-BetterGui

Fix bug cannot wrap text

josepholim committed:
Add gui package

josepholim committed:
Add more personality

josepholim committed:
Merge pull request #6 from josepholim/branch-A-Personality

Add more personality

josepholim committed:
Add error handling

josepholim committed:
Merge pull request #7 from josepholim/branch-A-MoreErrorHandling

Add error handling

josepholim committed:
Add More Testing

josepholim committed:
Merge pull request #8 from josepholim/branch-A-MoreTesting

Add More Testing

josepholim committed:
Update README.md

josepholim committed:
Add screenshot

josepholim committed:
Fix Event comparator

josepholim committed:
Fix Bye Command

josepholim committed:
Update README.md

josepholim committed:
Update README.md

josepholim committed:
Fix unknown command auto exit

josepholim committed:
Update README.md

josepholim committed:
Remove unnecessary this keyword

josepholim committed:
Add latest release

josepholim committed:
Add dependencies

josepholim committed:
Merge branch 'master' of https://github.com/josepholim/ip
